controlplane/src/core/repositories/analytics/TraceRepository.ts

```diff
@@ -54,9 +42,34 @@ export class TraceRepository {
         SpanAttributes['wg.router.version'] as attrRouterVersion,
         SpanAttributes['wg.operation.persisted_id'] as attrOperationPersistedId,
         SpanAttributes['wg.federated_graph.id'] as attrFederatedGraphId
-    FROM ${this.client.database}.otel_traces
-    WHERE (TraceId = trace_id) AND (Timestamp >= start) AND (Timestamp <= end) AND SpanAttributes['wg.organization.id'] = '${organizationID}'
-    ORDER BY Timestamp ASC
+    `;
+
+    const query = `
+    WITH RECURSIVE spans AS (
+      SELECT ${columns}
+      FROM otel_traces
+      WHERE (TraceId = trace_id) AND (Timestamp >= start) AND (Timestamp <= end) AND SpanAttributes['wg.organization.id'] = '${organizationID}' AND spanId = '${spanID}'
```

Comment:
We should also filter on graph id right?

Reply:
Yes, fixed